dkc/core/rest/file.py

```diff
@@ -71,10 +91,21 @@ class Meta(FileSerializer.Meta):
         read_only_fields = FileSerializer.Meta.read_only_fields + ['folder', 'size']
 
 
+class CreateWithAuthorizedUpload(BasePermission):
+    def has_object_permission(self, request, view, obj) -> bool:
+        return False
+
+    def has_permission(self, request: Request, view: View) -> bool:
+        # This doesn't actually do the policy enforcement, it simply allows requests
+        # through if they are authorized upload requests. Enforcement is done in the
+        # view's `perform_create` method.
+        return view.action == 'create' and 'authorization' in request.data
```

Comment:
I just ran into a security bug that makes it clear that we need to move all permission checking logic out of `perform_create` and into the serializer. As written, someone can pass any `authorization` value, and other validators will be run before the authorization is cryptographically verified. This is problematic because it can leak things like existence of a file name in a folder that a user doesn't have any access to.

Reply:
Note that this same issue already existed in the POST /folders endpoint, wherein permission checking is done inside `perform_create`. I'll add a ticket to address that.